tests/utils/test_config.py

```diff
@@ -93,3 +94,61 @@ def test_from_configuration_files_get_typed_value(tmp_path_factory: pytest.TempP
 
     assert Config().get_bool("legacy-current-snapshot-id")
     assert Config().get_int("max-workers") == 4
+
+
+@pytest.mark.parametrize(
+    "config_location, config_content, expected_result",
```

Comment:
nit: config_content and expected_result are always the same, we can use one of them, write to the file and verify that it reads back the same